test/scale/scale_tester.go

```diff
@@ -0,0 +1,292 @@
+	CSIDriver              driver.DynamicPVTestDriver
+	Pod                    testtypes.PodDetails
+	Replicas               int
+	StorageClassParameters map[string]string
+}
+
+type PodSchedulingOnFailoverScaleTest struct {
+	CSIDriver              driver.DynamicPVTestDriver
+	Pod                    testtypes.PodDetails
+	PodCount               int
+	Replicas               int
+	StorageClassParameters map[string]string
+}
+
+func (t *PodSchedulingWithPVScaleTest) Run(client clientset.Interface, namespace *v1.Namespace, schedulerName string) {
+	totalNumberOfPods := t.Replicas
+	var oneMinPodsRunning,
+		twoMinPodsRunning,
+		threeMinPodsRunning,
+		fiveMinPodsRunning,
+		tenMinPodsRunning = totalNumberOfPods, totalNumberOfPods, totalNumberOfPods, totalNumberOfPods, totalNumberOfPods
+	var oneMinPodsDeleted,
+		twoMinPodsDeleted,
+		threeMinPodsDeleted,
+		fiveMinPodsDeleted,
+		tenMinPodsDeleted = totalNumberOfPods, totalNumberOfPods, totalNumberOfPods, totalNumberOfPods, totalNumberOfPods
+
+	startScaleOut := time.Now()
+	tStorageClass, storageCleanup := t.Pod.CreateStorageClass(client, namespace, t.CSIDriver, t.StorageClassParameters)
+	defer storageCleanup()
+	statefulSet, cleanupStatefulSet := t.Pod.SetupStatefulset(client, namespace, t.CSIDriver, schedulerName, t.Replicas, t.StorageClassParameters, &tStorageClass)
+	for i := range cleanupStatefulSet {
+		i := i
+		defer cleanupStatefulSet[i]()
+	}
+	statefulSet.CreateWithoutWaiting()
+	numberOfRunningPods := 0
+	ticker := time.NewTicker(time.Minute)
+	tickerStop := make(chan bool)
+	go func() {
+		time.Sleep(11 * time.Minute)
+		tickerStop <- true
+	}()
+	tickerCount := 0
+	stopLoop := false
+	for {
+		select {
+		case <-tickerStop:
+			stopLoop = true
+
+		case <-ticker.C:
+			tickerCount = tickerCount + 1
+			numberOfRunningPods = podutil.CountAllPodsWithMatchingLabel(client, namespace, map[string]string{"app": "azuredisk-volume-tester"}, string(v1.PodRunning))
+			if numberOfRunningPods >= totalNumberOfPods {
+				stopLoop = true
+			}
+
+			if tickerCount == 1 {
+				oneMinPodsRunning = numberOfRunningPods
+			} else if tickerCount == 2 {
+				twoMinPodsRunning = numberOfRunningPods
+			} else if tickerCount == 3 {
+				threeMinPodsRunning = numberOfRunningPods
+			} else if tickerCount == 5 {
+				fiveMinPodsRunning = numberOfRunningPods
+			} else if tickerCount == 10 {
+				tenMinPodsRunning = numberOfRunningPods
+			}
+		}
+
+		if stopLoop {
+			break
+		}
+	}
+
+	scaleOutCompleted := time.Since(startScaleOut).Minutes()
+
+	startScaleIn := time.Now()
+	newScale := &scale.Scale{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      statefulSet.Statefulset.Name,
+			Namespace: statefulSet.Namespace.Name,
+		},
+		Spec: scale.ScaleSpec{
+			Replicas: int32(0),
+		},
+	}
+	_, _ = client.AppsV1().StatefulSets(statefulSet.Namespace.Name).UpdateScale(context.TODO(), statefulSet.Statefulset.Name, newScale, metav1.UpdateOptions{})
+
+	numberOfRemainingPods := totalNumberOfPods
+	tickerCount = 0
```

Comment:
You'll need to call [`Reset`](https://pkg.go.dev/time#Ticker.Reset) on the ticker after scaling down to ensure the next tick will be minute later.